tslearn/nonmyopic.py

```diff
@@ -0,0 +1,509 @@
+                ].predict(xt[:, :time_prediction])
+                result_proba = self.classifier_[
+                    time_prediction - self.minimum_time_stamp
+                ].predict_proba(xt[:, :time_prediction])
+                loss_exit = minimum_loss
+                stop_criterion = True
+            else:
+                time_prediction = time_prediction + 1
+        return result, time_prediction, result_proba, loss_exit
+
+    @staticmethod
+    def avg_prediction(time_predictions):
+
+        avgerage_time = sum(time_predictions) / len(time_predictions)
+        variance_time = np.var(time_predictions)
+        return avgerage_time, variance_time
+
+    @staticmethod
+    def accuracy_prediction(Y_test, Y_pred_test):
+
+        good_prediction = 0
+        for class_index in range(0, len(Y_test)):
+            if Y_test[class_index] == Y_pred_test[class_index]:
+                good_prediction = good_prediction + 1
+        return good_prediction / len(Y_test)
+
+    def predict(self, X):
+        """
+        Predicts the classes of the series of the test dataset.
+
+        Parameters
+        ----------
+        X_test : Array-like
+            The test dataset
+
+        Returns
+        -------
+        Vector : the predicted classes
+        """
+
+        Y_pred = []
+
+        for t in range(0, X.shape[0]):
+            new_classif, new_time, new_proba, new_cost = self.classification(
+                X[t], self.cost_function
+            )
+            Y_pred.append(new_classif)
+        return Y_pred
+
+    def predict_completed(self, X, Y):
+        """
+        Predicts the classes of the series of the test dataset and has also access the the true classes to establish
+        comparisons
+
+        Parameters
+        ----------
+        X: Array-like
+            The data set that needs to be predict
+        Y: Array-vector
+            The true classes
+
+        Returns
+        -------
+        float : average time of prediction
+        float : variance of the time of prediction
+        float : AUC
+        float : average cost
+        """
+        Y_pred = []
+        time_predictions = []
+        pred_proba = []
+        cost_average = 0
+
+        for current_serie in range(0, X.shape[0]):
+            new_classif, new_time, new_proba, new_cost = self.classification(
+                X[current_serie], self.cost_function
+            )
+            Y_pred.append(new_classif)
+            time_predictions.append(new_time)
+            pred_proba.append(new_proba)
+            cost_average = cost_average + new_cost
+        average_pred, variance_pred = self.avg_prediction(time_predictions)
+        pred_proba = np.reshape(np.asarray(pred_proba), (X.shape[0], 2))
+        auc = roc_auc_score(Y, pred_proba[:, 1])
+        cost_average = cost_average / X.shape[0]
+        return average_pred, variance_pred, auc, cost_average
+
+    def graphic(self, path, name, prediction_time=None):
```

Comment:
This should not be included in the class. This is code you use for visualization but the class code should just consist in what is required to build your estimator, fit it to data, and predict outcome for new data.